lib/Proxy/AbstractProxy.php

```diff
@@ -42,46 +44,56 @@ abstract class AbstractProxy
      * initializes and runs the proxy class
      *
      * @access public
+     * @param Configuration $conf
      * @param string $link
      */
-    public function __construct(Configuration $conf, $link)
+    public function __construct(Configuration $conf, string $link)
     {
         if (!str_starts_with($link, $conf->getKey('basepath') . '?')) {
             $this->_error = 'Trying to shorten a URL that isn\'t pointing at our instance.';
             return;
         }
 
-        $data = $this->_getcontents($conf, $link);
+        $proxyUrl = $this->_getProxyUrl($conf);
 
-        if ($data == null) {
-            $this->_error = 'Error calling proxy. Probably a configuration issue';
-            error_log('Error calling proxy: ' . $this->_error);
+        if (empty($proxyUrl)) {
+            $this->_error = 'Error calling proxy. Probably a configuration issue, like wrong or missing config keys.';
```

Comment:
The proxy has not actually been called here, so we can be more exact…

```suggestion
            $this->_error = 'Proxy error: Proxy URL is empty. This is probably a configuration issue, like wrong or missing config keys.';
```